lib/src/ui/manup_widget.dart

```diff
@@ -0,0 +1,51 @@
+part of manup;
+
+class ManUpWidget extends StatefulWidget {
+  final ManUpService service;
+  final Widget child;
+  final bool Function() shouldShowAlert;
+  final void Function(bool) onComplete;
+  final void Function(dynamic) onError;
+  //
+  ManUpWidget(
+      {Key key,
+      @required this.child,
+      @required this.service,
+      @required this.shouldShowAlert,
+      @required this.onComplete,
+      @required this.onError})
+      : super(key: key);
+
+  @override
+  _ManUpWidgetState createState() => _ManUpWidgetState();
+}
+
+class _ManUpWidgetState extends State<ManUpWidget>
+    with ManupDelegate, ManupDelegateMixin {
```

Comment:
I feel like the `ManupDelegate` and `ManupDelegateMixin` need to be two totally separate things (rather than one being an extension on the other). Separate the delegate as "a thing that responds to manup events" and just merge the rest back into the `ManupService`.

Reply:
ManupDelegateMixin is default implementation so we dont need to implement all methods. it is not required to implement just ManupDelegate is enough.

Reply:
I think in the spirit of ManUp - we want this to be as 'drop-in' as possible, while also allowing users to customise the functionality themselves if they wanted.

Should we have a ManUp widget that wraps the main app body, and defines the delegate methods etc at this level? This would then give the service the context it requires, as well as allowing for that easy predefined drop-in behaviour?